pkg/controller/node/nodecontroller.go

```diff
@@ -551,20 +564,14 @@ func (nc *NodeController) monitorNodeStatus() error {
 			// Check eviction timeout against decisionTimestamp
 			if observedReadyCondition.Status == api.ConditionFalse &&
 				decisionTimestamp.After(nc.nodeStatusMap[node.Name].readyTransitionTimestamp.Add(nc.podEvictionTimeout)) {
-				if nc.evictPods(node) {
-					glog.V(4).Infof("Evicting pods on node %s: %v is later than %v + %v", node.Name, decisionTimestamp, nc.nodeStatusMap[node.Name].readyTransitionTimestamp, nc.podEvictionTimeout)
-				}
+				addNodeOutageTaint(nc.kubeClient, node)
```

Comment:
I think we have a misunderstanding about maintenance mode. I had no plans to use node-maintenance:NoExecute. My use-case is precisely this one - https://github.com/kubernetes/kubernetes/issues/30272#issuecomment-243069603, e.g. when node will stay in healthy state, but kubelet itself will be under upgrade procedure. Please take a look at this comment https://github.com/kubernetes/kubernetes/issues/30272#issuecomment-250108306.

We can actually cover both use cases from kubectl side:

1. don't schedule new, but don't evict existing
2. don't schedule new and evict existing pods before maintenance is started

In first case user will have to assign NoSchedule taint and change forgiveness to NoExecute taint. In second - assign NoSchedule, remove all pods and wait until all of them will be rescheduled somewhere else. What do you think?